### PR TITLE
SDK-1910 Doc Scan Watchlist Advanced Ca Check

### PR DESCRIPTION
**Pull request by laurent-yoti**
#### Part1: Session creation
Added WatchListAdvancedCa session creation parts: Configs, Check and Builders

#### Part2: Session retrieval
Added WatchListAdvancedCa session retrieval parts: Responses

#### Part3: Example
Added support for Watchlist advanced checks.

screencapture-localhost-3000-success-2021-08-04-11_39_17

### CODE REVIEW

File: examples/doc-scan/views/pages/partials/check.ejs [hunk 1]
```diff
@@ -1,5 +1,12 @@
-<% displayGeneratedProfile = typeof hasGeneratedProfile === 'undefined' ? false : hasGeneratedProfile %>
-<% displayWatchlistSummary = typeof hasWatchlistSummary === 'undefined' ? false : hasWatchlistSummary %>
+<%
+formatString = function (value) {
+    if(value) {
+        const stringValue = value.toString()
+        return stringValue.charAt(0).toUpperCase() + stringValue.slice(1).replace(/_/g, ' ').replace(/-/g, ' ').toLowerCase()
```
Comment (laurent-yoti) [inline examples/doc-scan/views/pages/partials/check.ejs:5]: So, it capitalises and replaces `_` and `-` with a space.
One could say that you could break it down in 2 functions, and if you need still have an extra function  which combines both.
Regarding the replace bit, you also combine into one call with the `|` operator, aka using `/_|-/g` for pattern.

File: examples/doc-scan/views/pages/success.ejs [hunk 2]
```diff
@@ -206,7 +206,29 @@
...
+                                data-target="#collapse-watch-list-advanced-ca-checks" aria-expanded="true"
+                                aria-controls="collapse-watch-list-advanced-ca-checks">
+                                Watchlist Advanced Checks
+                            </button>
+                        </h3>
+                    </div>
+
+                    <div id="collapse-watch-list-advanced-ca-checks" class="collapse" aria-labelledby="collapse-watch-list-advanced-ca-checks">
+                        <div class="card-body">
+                            <% sessionResult.getWatchlistAdvancedCaChecks().forEach(function(check){ %>
```
Comment (laurent-yoti) [inline examples/doc-scan/views/pages/success.ejs:230]: Not sure I understand why there would be several advanced checks, given you are only requesting one, what am I missing ?

File: examples/doc-scan/views/pages/partials/check.ejs (hunk 1 above)
Comment (dhivya-pattabhiraman) [reply examples/doc-scan/views/pages/partials/check.ejs]: Addressed this one.

File: examples/doc-scan/views/pages/success.ejs (hunk 2 above)
Comment (dhivya-pattabhiraman) [reply examples/doc-scan/views/pages/success.ejs:230]: That is sort of the pattern followed everywhere.

Comment (echarrod) [reply examples/doc-scan/views/pages/success.ejs:230]: Yep, there might be further checks added in the future, and if it's left like this then we won't need to update the code :slightly_smiling_face:

File: src/doc_scan_service/session/retrieve/watchlist.summary.response.js [hunk 3]
```diff
@@ -1,11 +1,16 @@
 'use strict';
 
-const RawResultsResponse = require('./raw.results.response');
-const SearchConfigResponse = require('./search.config.response');
 const Validation = require('../../../yoti_common/validation');
+const RawResultsResponse = require('./raw.results.response');
+// const WatchlistSearchConfigResponse = require('./watchlist.search.config.response');
```
Comment (echarrod) [inline src/doc_scan_service/session/retrieve/watchlist.summary.response.js:5]:
```suggestion
```

File: tests/doc_scan_service/session/retrieve/watchlist.screening.summary.response.spec.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+// eslint-disable-next-line max-len
+// const WatchlistSummaryResponse = require('../../../../src/doc_scan_service/session/retrieve/watchlist.summary.response');
```
Comment (echarrod) [inline tests/doc_scan_service/session/retrieve/watchlist.screening.summary.response.spec.js:2]:
```suggestion
```

File: tests/doc_scan_service/session/retrieve/custom.account.watchlist.ca.search.config.response.spec.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+          expect(sources).toBeInstanceOf(TypeListSourcesResponse);
+          expect(sources.getType()).toBe('TYPE_LIST');
+          expect(sources.getTypes()).toStrictEqual(['type1', 'type2']);
+        });
+      });
+    });
+    describe('with matching strategy in response body', () => {
+      describe('of type EXACT', () => {
+        beforeEach(() => {
+        // eslint-disable-next-line max-len
```
Comment (echarrod) [inline tests/doc_scan_service/session/retrieve/custom.account.watchlist.ca.search.config.response.spec.js:124]: Guessing we don't need this any more
```suggestion
```

**Review by echarrod — APPROVED** [review]
A few lines that can be removed, looks good otherwise! :+1:

File: src/doc_scan_service/session/retrieve/watchlist.summary.response.js (hunk 3 above)
Comment (laurent-yoti) [reply src/doc_scan_service/session/retrieve/watchlist.summary.response.js]: 👍

File: tests/doc_scan_service/session/retrieve/custom.account.watchlist.ca.search.config.response.spec.js (hunk 5 above)
Comment (laurent-yoti) [reply tests/doc_scan_service/session/retrieve/custom.account.watchlist.ca.search.config.response.spec.js]: 👍 yeah, there were a couple of those, annoying as they stick around after refactoring, it's almost that you'd need a rule in eslint to warn that you are using a non necessary `eslint-disable`...

**Pull request merged**